src/Region.h

```diff
@@ -0,0 +1,172 @@
+/**
+ * @file Region.h
+ * @brief The Region class.
+ * @date March 10, 2017
+ * @author William Boyd, MIT, Course 22 (wboyd@mit.edu)
+ */
+ #ifndef REGION_H_
+#define REGION_H_
+ #ifdef __cplusplus
+#ifdef SWIG
+#include "Python.h"
+#endif
+#include "Surface.h"
+#include "boundary_type.h"
+#include <limits>
+#endif
+
+ /* Forward declarations to resolve circular dependencies */
+class Intersection;
+class Union;
+class Complement;
+class Halfspace;
+
+/**
+ * @enum regionType
+ * @brief The types of regions supported by OpenMOC.
+ */
+enum regionType {
+  /** The intersection of one or more regions */
+  INTERSECTION,
+   /** The union of one or more regions */
+  UNION,
+   /** The complement of a region */
+  COMPLEMENT,
+   /** The side of a surface */
+  HALFSPACE
+};
+
+/**
+ * @class Region Region.h "src/Region.h"
+ * @brief A region of space that can be assigned to a Cell.
+ */
+class Region {
+
+protected:
+
+   /** The type of Region (ie, UNION, INTERSECTION, etc) */
+  regionType _region_type;
+
+   /** A collection of the nodes within the Region */
+  std::vector<Region*> _nodes;
+
+  /** The parent region, a region which has this region among its nodes */
+  Region* _parent_region;
```

Comment:
How about adding a constructor to set the default _region_type(INTERSECTION) and _parent_region(NULL)?

Reply:
Added